### PR TITLE
Upgrade JS dependencies

### PR DESCRIPTION
**Pull request by ahangarha**
In this PR, I have updated our JS dependencies to address several known security vulnerabilities. Amoung them and due to our urgency I resolved the issues with `ansi-regex` using `resolutions` entry in `yarn.lock`. This is a temporary fix due to complexities of upgrading eslint and other relevant dependencies (namely prettier-eslint-cli) to newer versions.

This change is [Reviewable](https://reviewable.io/reviews/shakacode/react_on_rails/1514)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -17,6 +17,9 @@ Changes since last non-beta release.
 
 *Please add entries here for your pull requests that are not yet released.*
 
+### Fixed
+- Upgrade several JS dependencies to knows security issues. [PR 1514](https://github.com/shakacode/react_on_rails/pull/1514) by [ahangarha](https://github.com/ahangarha)
```
Comment (justin808) [inline CHANGELOG.md]: Upgrade several JS dependencies to fix security issues.

File: package.json [hunk 2]
```diff
@@ -53,6 +54,9 @@
     "react": ">= 16",
     "react-dom": ">= 16"
   },
+  "resolutions": {
+    "ansi-regex": "^3.0.1"
```
Comment (justin808) [inline package.json]: why is this needed?

should be documented in the PR

File: package.json [hunk 3]
```diff
@@ -7,42 +7,43 @@
...
     "jsdom": "^16.4.0",
     "nps": "^5.9.3",
-    "prettier": "^2.2.1",
+    "prettier": "^2.8.3",
     "prettier-eslint-cli": "^5.0.0",
-    "prop-types": "^15.5.10",
-    "react": "^16.5.2",
-    "react-dom": "^16.5.2",
+    "prop-types": "^15.8.1",
+    "react": "^16.14.0",
```
Comment (justin808) [inline package.json:41]: Why is react not upgraded to 17 at least?

Comment (ahangarha) [reply package.json:41]: There was no security issue with React. For such dependencies, I only pushed all minor versions. I thought it might be a development decision to use which version of React.

File: package.json (hunk 2 above)
Comment (ahangarha) [reply package.json]: I explained in its [own commit](https://github.com/shakacode/react_on_rails/pull/1514/commits/a019f7e03a1dd5e8bfbc747549ae05ef60a7ff10):

> This is a temporary fix due to complexities of upgrading eslint and
> other relevant dependencies (namely prettier-eslint-cli) to newer
> versions.

Maybe I had to bring it to the PR body as well.

Since we had a kind of urgency in resolving high/critical vulnerabilities, I did a tradeoff and chose to use this approach for this particular package rather than spending hours or days to fix issues with upgrading `prettier-eslint-cli`, which required upgrading `eslint` and make modifications in our source code to address the breaking changes.

File: CHANGELOG.md (hunk 1 above)
Comment (ahangarha) [reply CHANGELOG.md]: My bad! Editing several times and finally pushing a wrong sentence! 😞 
Fixed ✅

File: package.json (hunk 2 above)
Comment (ahangarha) [reply package.json]: Fixed ✅
